snap-core/src/main/java/org/esa/snap/core/util/GeoUtils.java

```diff
@@ -0,0 +1,639 @@
+     */
+    public static GeoPos[] createGeoBoundary(Product product, Rectangle region, int step,
+                                             final boolean usePixelCenter) {
+        final GeoCoding gc = product.getSceneGeoCoding();
+        if (gc == null) {
+            throw new IllegalArgumentException(UtilConstants.MSG_NO_GEO_CODING);
+        }
+
+        if (region == null) {
+            region = new Rectangle(0,
+                    0,
+                    product.getSceneRasterWidth(),
+                    product.getSceneRasterHeight());
+        }
+
+        return getGeoBoundary(gc, region, step, usePixelCenter);
+    }
+
+    /**
+     * Creates the geographical boundary of the given region within the given raster and returns it as a list of
+     * geographical coordinates.
+     *
+     * @param raster the input raster, must not be null
+     * @param region the region rectangle in raster pixel coordinates, can be null for entire raster
+     * @param step   the step given in pixels
+     * @return an array of geographical coordinates
+     * @throws IllegalArgumentException if raster is null or if the raster has no {@link GeoCoding} is null
+     * @see #createPixelBoundary(RasterDataNode, java.awt.Rectangle, int)
+     */
+    public static GeoPos[] createGeoBoundary(RasterDataNode raster, Rectangle region, int step) {
+        return createGeoBoundary(raster, region, step, true);
+    }
+
+    /**
+     * Creates the geographical boundary of the given region within the given RasterDataNode and returns it as a list of
+     * geographical coordinates.
+     *
+     * @param rasterDataNode the input rasterDataNode, must not be null
+     * @param region         the region rectangle in product pixel coordinates, can be null for entire product
+     * @param step           the step given in pixels
+     * @param usePixelCenter {@code true} if the pixel center should be used to create the boundary
+     * @return an array of geographical coordinates
+     * @throws IllegalArgumentException if product is null or if the product's {@link GeoCoding} is null
+     */
+    public static GeoPos[] createGeoBoundary(RasterDataNode rasterDataNode, Rectangle region, int step,
+                                             final boolean usePixelCenter) {
+        final GeoCoding gc = rasterDataNode.getGeoCoding();
+        if (gc == null) {
+            throw new IllegalArgumentException(UtilConstants.MSG_NO_GEO_CODING);
+        }
+
+        if (region == null) {
+            region = new Rectangle(0,
+                    0,
+                    rasterDataNode.getRasterWidth(),
+                    rasterDataNode.getRasterHeight());
+        }
+
+        return getGeoBoundary(gc, region, step, usePixelCenter);
+    }
+
+    // @todo 1 tb/tb implement this! 2020-01-31
+//    public static ArrayList<GeoPos[]> createGeoBoundaries(Product product, Rectangle region, int step,
```

Comment:
Do we need this method? Isn't it covered by 
`GeneralPath[] createGeoBoundaryPaths(Product product, Rectangle region, int step, final boolean usePixelCenter)`?

Reply:
This is a planned API method to allow for segmented geo-bounds, i.e. geo-boundaries that consist of multiple segments separated by fill-value regions. We should discuss this separately, that is a non-trivial task to integrate this into SNAP

Reply:
Sure, okay!